### PR TITLE
Test : Module List storage tests

### LINKED ISSUE #548: Tests : Update Junit tests for storage component
Include Junit testing for module list storage

### PR DESCRIPTION
**Pull request by davidcaiqifan**
Fixes #548

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedModule.java [hunk 1]
```diff
@@ -113,6 +113,11 @@ public Module toModelType() throws IllegalValueException {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
                     GradeTracker.class.getSimpleName()));
         }
+        try {
+            gradeTracker.toModelType();
+        } catch (IllegalValueException illegalValueException) {
```
Comment (jonasngs) [inline src/main/java/seedu/address/storage/JsonAdaptedModule.java:118]: I think catching and throwing the same exception here is redundant, as long as the method caller is handling the correct exception, we can omit this try catch block.

**Review by jonasngs — APPROVED** [review]
LGTM.

**Review by murtubak — APPROVED** [review]
LGTM. Don't see anything that needs to be changed.

File: src/main/java/seedu/address/storage/JsonAdaptedModule.java (hunk 1 above)
Comment (davidcaiqifan) [reply src/main/java/seedu/address/storage/JsonAdaptedModule.java]: Thanks, noted!

**Pull request merged**